en/docs/develop/writing-a-custom-event-handler.md

```diff
@@ -1,228 +1,160 @@
-          Entity[] entities = new Entity[1];
-          entities[0] = new Entity(serviceProvider.getApplicationName(), "SP", -1234);
-          if (!Boolean.TRUE.equals(getWorkFlowCompleted()) && !isValidOperation(entities)) {
-              throw new WorkflowException("Operation is not valid.");
-          }
-          boolean state = startWorkFlow(wfParams, nonWfParams, uuid).getExecutorResultState().state();
-          return state;
-      }
-
-      @Override
-      public boolean isValidOperation(Entity[] entities) throws WorkflowException {
-          //Check if the operation is valid, eg:- Is there a SP already added and not approved with the same name as
-          // this SP.
-          return true;
-      }
-    }
+The WSO2 Identity Server eventing framework can be used to trigger some events such as user operations. The eventing framework also supports handlers which can be used to do some operations based on the published events.
+
+## What is an event handler?
+
+An event handler is used to perform an operation based on the published events. 
+
+For instance, an event handler can be used to send an email after a user addition. The following sequence of operations are executed while adding a user.
+
+1. Publish the `PRE_ADD_USER` event — The subscribed handlers will be executed for the pre-add user event.
+2. Execute the `AddUser` operation — The user will be persisted to the user store (LDAP or JBDC user store).
+3. Publish the `POST_ADD_USER` event — The subscribed handlers will be executed for the post-add user event.
+
+Therefore, the email can be sent through an event handler that is subscribed to the `POST_ADD_USER` event.
+
+The following list is a list of sample events. 
+
+- PRE_AUTHENTICATION
+- POST_AUTHENTICATION
+- PRE_SET_USER_CLAIMS
+- POST_SET_USER_CLAIMS
+- PRE_ADD_USER
+- POST_ADD_USER
+- PRE_UPDATE_CREDENTIAL
+- POST_UPDATE_CREDENTIAL
+- PRE_UPDATE_CREDENTIAL_BY_ADMIN
+- POST_UPDATE_CREDENTIAL_BY_ADMIN
+- PRE_DELETE_USER
+- POST_DELETE_USER
+- PRE_SET_USER_CLAIM
+- PRE_GET_USER_CLAIM
+- POST_GET_USER_CLAIMS
+- POST_GET_USER_CLAIM
+- POST_SET_USER_CLAIM
+- PRE_DELETE_USER_CLAIMS
+- POST_DELETE_USER_CLAIMS
+- PRE_DELETE_USER_CLAIM
+- POST_DELETE_USER_CLAIM
+- PRE_ADD_ROLE
+- POST_ADD_ROLE
+- PRE_DELETE_ROLE
+- POST_DELETE_ROLE
+- PRE_UPDATE_ROLE
+- POST_UPDATE_ROLE
+- PRE_UPDATE_USER_LIST_OF_ROLE
+- POST_UPDATE_USER_LIST_OF_ROLE
+- PRE_UPDATE_ROLE_LIST_OF_USER
+- POST_UPDATE_ROLE_LIST_OF_USER
+- UPDATE_GOVERNANCE_CONFIGURATION
+- TRIGGER_NOTIFICATION
+
+## Writing an event handler
+
+To write a new event handler, you must extend the `org.wso2.carbon.identity.event.handler.AbstractEventHandler`. 
+
+1. Initialize the events in the [Events class](https://github.com/wso2/carbon-identity-framework/blob/0cdd0641a6dd4ece47c79ada5acc05a5d7f9bbe0/components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConstants.java#L47) as follows.
```

Comment:
This can go in to the publishing custom event and handle that in the handler section.

Reply:
On second thought, let's remove the initializing events section which is irrelevant to writing an event handler.